src/psyclone/tests/dynamo0p3_test.py

```diff
@@ -306,22 +306,6 @@ def test_unnecessary_shape():
             in str(excinfo.value))
 
 
-def test_kernel_call_invalid_iteration_space():
```

Comment:
I guess this one is covered by the metadata tests so isn't needed? Do we therefore still need the Fortran source files it uses or should we keep them for when we do support a user-supplied kernel that operates on 'dofs'?

Reply:
In fact, if you got to https://app.codecov.io/gh/stfc/PSyclone/pull/2675/indirect-changes you'll see that we are now missing two lines that were previously covered. (It would be good if CodeCov highlighted this in its comments on the PR but hey, that's what the reviewer is for :-) )

Reply:
I think we should keep them for the next step in this little dofkern series - code generation. Having added back the `gen_stub` tests, the only line uncovered is in `validate_global_constraints`. I'll add this test back with an iteration space other than 'dof' as the test case.